docker-compose.yaml

```diff
@@ -0,0 +1,15 @@
+---
+version: "3.8"
+
+services:
+  ollama:
+    build: .
+    container_name: ollama
+    image: ollama/ollama
+    healthcheck:
+      test: curl --fail http://localhost:11434 || exit 1
```

Comment:
For some reason, this works outside of the container, but in the healthcheck it's not working. Anyone have an idea why not?

Reply:
Ah figured it out, `curl` isn't present in the image

Reply:
Any idea how to do this healthcheck using shell commands, since `curl` isn't present?

I moved to `ollama --version` for now, please advise if there's any better idea.  Cc @BruceMacD since you're

Reply:
Hmm, I'm not a specialist on shell commands by any means, but calling the API directly would probably be better if possible. To see that Ollama is ready to receive requests rather than Ollama being installed. I wonder if `wget` would work.

Reply:
Nice thinking on `wget`, but it's not present either, and neither is `python`:

```none
> docker compose exec ollama bash
root@928481477b63:/# wget
bash: wget: command not found
root@928481477b63:/# curl
bash: curl: command not found
root@928481477b63:/# python
bash: python: command not found
```

Any other ideas? I see `perl-base` is installed, I wonder if there's some sense in writing a Perl script for the health check.

Reply:
maybe you can add (`apt install curl`) on a step at Dockerfile or at docker compose (via `command: ...`)? - but I'm not a fan

Reply:
Okay so it turns out `LWP/Simple.pm` is not in the container either:

```bash
> docker run -it --entrypoint bash ollama/ollama
> root@0579a3e6a13b:/# perl healthcheck.pl
Can't locate LWP/Simple.pm in @INC (you may need to install the LWP::Simple module) (@INC contains: /etc/perl /usr/local/lib/aarch64-linux-gnu/perl/5.34.0 /usr/local/share/perl/5.34.0 /usr/lib/aarch64-linux-gnu/perl5/5.34 /usr/share/perl5 /usr/lib/aarch64-linux-gnu/perl-base /usr/lib/aarch64-linux-gnu/perl/5.34 /usr/share/perl/5.34 /usr/local/lib/site_perl) at healthcheck.pl line 1.
BEGIN failed--compilation aborted at healthcheck.pl line 1.
root@0579a3e6a13b:/# perl -MCPAN -e'install "LWP::Simple"'
```

Reply:
@jmorganca @BruceMacD @technovangelist are you open to me adding `wget` to the production image to enable [`HEALTHCHECK`](https://docs.docker.com/engine/reference/builder/#healthcheck)?

Reply:
See https://github.com/jmorganca/ollama/pull/1909, I upstreamed to `Dockerfile` using `curl`

Reply:
Alright #1909 (which includes adding `curl` to the base image) was not accepted (see discussion in that PR), so that defeats the `curl` approach.

However, looks like @stevenbecht in https://github.com/ollama/ollama/pull/2045 poses a way to use bash sockets to accomplish the healthcheck. TIL of bash sockets! I will adopt that technique